packages/nitro-protocol/contracts/ForceMove.sol

```diff
@@ -52,7 +52,7 @@ contract ForceMove is IForceMove {
         Signature[] memory sigs,
         uint8[] memory whoSignedWhat,
         Signature memory challengerSig
-    ) public override {
+    ) external override {
```

Comment:
Point of confusion. This contract implements the `IForceMove` interface, where `external` method parameters have their parameters marked as `calldata` instead of `memory`. 

If I make that change in *this* contract, I get 
```shell
➜  nitro-protocol git:(externalize) ✗ yarn contract:compile --show-stack-traces
yarn run v1.22.5
$ yarn hardhat compile --force --show-stack-traces
$ /Users/georgeknee/statechannels/statechannels/node_modules/.bin/hardhat compile --force --show-stack-traces
Compiling 25 files with 0.7.4
UnimplementedFeatureError: 

Error HH600: Compilation failed

HardhatError: HH600: Compilation failed
    at SimpleTaskDefinition.action (/Users/georgeknee/statechannels/statechannels/node_modules/hardhat/src/builtin-tasks/compile.ts:741:15)
    at Environment._runTaskDefinition (/Users/georgeknee/statechannels/statechannels/node_modules/hardhat/src/internal/core/runtime-environment.ts:217:14)
    at Environment.run (/Users/georgeknee/statechannels/statechannels/node_modules/hardhat/src/internal/core/runtime-environment.ts:130:14)
    at SimpleTaskDefinition.action (/Users/georgeknee/statechannels/statechannels/node_modules/hardhat/src/builtin-tasks/compile.ts:954:7)
    at Environment._runTaskDefinition (/Users/georgeknee/statechannels/statechannels/node_modules/hardhat/src/internal/core/runtime-environment.ts:217:14)
    at Environment.run (/Users/georgeknee/statechannels/statechannels/node_modules/hardhat/src/internal/core/runtime-environment.ts:130:14)
    at SimpleTaskDefinition.action (/Users/georgeknee/statechannels/statechannels/node_modules/hardhat/src/builtin-tasks/compile.ts:384:45)
    at Environment._runTaskDefinition (/Users/georgeknee/statechannels/statechannels/node_modules/hardhat/src/internal/core/runtime-environment.ts:217:14)
    at Environment.run (/Users/georgeknee/statechannels/statechannels/node_modules/hardhat/src/internal/core/runtime-environment.ts:130:14)
    at SimpleTaskDefinition.action (/Users/georgeknee/statechannels/statechannels/node_modules/hardhat/src/builtin-tasks/compile.ts:1172:63)
error Command failed with exit code 1.
```

Reply:
Digging into this a bit more, I was starting to see errors like this

```
contracts/ForceMove.sol:479:17: TypeError: Invalid type for argument in function call. Invalid implicit conversion from struct IForceMove.FixedPart memory to struct IForceMove.FixedPart calldata requested.
                fixedPart,
```
and now things make sense. This PR didn't actually change any parameters from `memory` to `calldata` (exceptions being base contracts which were overridden anyway), hence no gas savings. Doing so causes problems when internal functions expect `memory` and not `calldata`. 

Making such changes can be attempted in separate PRs.